Fixed #195: Fixed fatal bug in tensorflow export

PalashTanejaPro opened the pull request:
This fix has been tested with my fork of caffe-tensorflow and tensorflow v1.4. 
It might not work on older tensorflow versions.

virajprabhu commented:
@PalashTanejaPro Thanks for the fix! @utsavgarg Can you please take a look?

utsavgarg commented:
@PalashTanejaPro It's a nice effort :+1: A few suggestions:
* GoogLeNet import is working fine with the Tensorflow backend, but it does not set all values properly so the export doesn't work
* We should avoid using `global` variables (used in `json_to_prototxt.py`)
* And the travis build is failing due to formatting errors, please correct that.

PalashTanejaPro commented:
- Export is working on my side.
![output](https://user-images.githubusercontent.com/22736920/34441315-dd9b7d48-ece0-11e7-8e85-2fec526205d5.gif)

- Removed global variable
- Fixed travis build

utsavgarg commented on ide/utils/jsonToPrototxt.py:
```diff
@@ -1320,12 +1328,17 @@ def changeTopBlobName(layerId, newName):
         layerType = layer['info']['type']
         layerPhase = layer['info']['phase']
+
+        if (str(layerType) == "Input"):
```
This doesn't necessarily have to be a layer of type `Input`.

utsavgarg commented on temp.json:
```diff
@@ -0,0 +1,10 @@
+{"l78": {"info": {"phase": null, "type": "Pooling"}, "shape": {"input": [512, 13, 13], "output": [512, 13, 13]}, "params": {"layer_type": "2D", "kernel_w": 3, "stride_d": 1, "pad_h": 1, "stride_h": 1, "pad_d": 0, "stride_w": 1, "kernel_d": "", "caffe": true, "kernel_h": 3, "pad_w": 1, "pool": "MAX"}, "connection": {"input": ["l67"], "output": ["l79"]}, "props": {"name": "l78"}}, "l79": {"info": { ...
```
You seem to have committed this file by mistake.

utsavgarg commented:
@PalashTanejaPro export is not working for me.
![ezgif com-video-to-gif 3](https://user-images.githubusercontent.com/11680179/34460878-cf19f8b4-ee41-11e7-85c0-2917bf53e69d.gif)

utsavgarg commented:
LGTM :)
